resources/lib/metadata.py

```diff
@@ -88,6 +88,16 @@ def get_context_menu(self, api_data, program, cache_file):
                         'RunPlugin(%s)' % self._kodi.url_for('follow', program=program, title=program_title)
                     ))
         context_menu.append((self._kodi.localize(30413), 'RunPlugin(%s)' % self._kodi.url_for('delete_cache', cache_file=cache_file)))
+
+        # Go to program context menu
+        if cache_file and cache_file.startswith(('offline', 'recent')):
```

Comment:
That's a clever way of doing this.

When investigating this, I gave up after realising it meant I had to forward even more information to helper functions and didn't think it was worth all the effort/complexity.